### DISCUSSION TITLE
Keep Exif Image Orientation on image processing

### DISCUSSION THREAD

**Discussion by dims31** [gethinode/hinode#611] [Ideas]
Hello,

It can be conveniant if Exif Image Orientation can be processed when rendering image. 

Many pictures taken with phones are upside down after image rendering and need manual correction before uploading.

**Comment by markdumay** [comment]
Would you mind sharing an example that demonstrates the issue? I would expect Hugo to recognize and process the EXIF data correctly.

**Reply by dims31** [reply]
Hello Mark,

This is an exemple with a photo taken phone upside down displayed correctly without image rendering.

[Exemple](https://deploy-preview-47--tourdesminous.netlify.app/destinations/indonesie/exif-issue/) 
[Source Image](https://dimstowp01.blob.core.windows.net/img/exif.jpg)
 [Github](https://github.com/dims31/tdm/blob/exif-issue/content/destinations/indonesie/exif-issue.md)


[Related Hugo Discussion](https://discourse.gohugo.io/t/image-exif-orientation/22902/7)

Thank you for your help !

**Comment by markdumay** [comment]
I added EXIF handling to [release v0.21.1](https://github.com/gethinode/hinode/releases/tag/v0.21.1). I used the images from this [gallery](https://www.galloway.me.uk/2012/01/uiimageorientation-exif-orientation-sample-images/) for testing.

I couldn't find a way for Hugo to mirror (horizontally flip) images, so I used CSS instead. However, rotation is done natively using Hugo's [image processing](https://gohugo.io/functions/images/filter/).

**Reply by dims31** [reply]
Thank you, this is working perfectly.

Now, I have a warning with pictures with EXIF metadata removed :

WARN  Unable to decode Exif metadata from image: /img/indonesiemunduk1.jpg

**Reply by markdumay** [reply]
Great to hear it's working. The warning is generated by Hugo itself. The exif library that Hugo uses isn't that great, but I'm not sure this is a priority for the maintainers. See https://github.com/gohugoio/hugo/issues/8996. I'll close this discussion for now.
